evm/contracts/ChainlinkClient.sol

```diff
@@ -194,6 +202,43 @@ contract ChainlinkClient {
     setChainlinkOracle(resolver.addr(oracleSubnode));
   }
 
+  /**
+   * @notice Determines whether the given address is a contract
+   * @param _addr The address to check
+   */
+  function isContract(address _addr)
+    private
+    view
+    returns (bool)
+  {
+    uint length;
+    assembly { length := extcodesize(_addr) }
+    return length > 0;
+  }
+
+  /**
+   * @notice Sets the Chainlink token and oracle address for
+   * the public network detected by the LINK token deployment
+   * @dev In order to save gas, we return early based on usage
+   * of the networks: Ropsten > Rinkeby > Kovan
+   */
+  function setChainlinkNetwork() internal {
```

Comment:
It would be nice if this could still be used on test networks too. Maybe we can have it take an address parameter, if that address is not zero use the parameter, otherwise try detecting the network.

Reply:
:+1: Updated description.